GOVERNANCE.md

```diff
@@ -7,12 +7,21 @@ This document defines project governance for the Strimzi project.
 ### Maintainers
 
 The maintainers are responsible for the overall development of the whole Strimzi project and all its components.
-The maintainers are identified in the [`MAINTAINERS`](MAINTAINERS) file. 
+Emeritus maintainers are former maintainers who are no longer actively involved in the project.
+In the rest of this document the term ‘maintainer’ will be assumed to mean ‘active maintainer’, unless qualified as ‘emeritus maintainer’.
+Both maintainers and emeritus maintainers are identified in the [`MAINTAINERS`](MAINTAINERS) file. 
 
 #### Changes in Maintainership
 
 New maintainers are proposed by an existing maintainer and are elected by a ⅔ majority maintainers vote.
-Maintainers can be removed by a ⅔ majority maintainers vote.
+Maintainers can be removed by a ⅔ majority maintainers vote. 
+Maintainers removed by a vote are no longer considered maintainers or emeritus maintainers.
+Maintainers become emeritus maintainers either actively, by opening a PR against the MAINTAINERS file marking themselves as Emeritus, or implicitly after 6 months without any contribution to the project.
+Updates to the MAINTAINERS to mark implicitly Emeritus maintainers happen asynchronously (i.e. the MAINTAINERS file is not normative about Emeritus status).
+An Emeritus maintainer reverts to being an active maintainer by contributing to the project again.
+For the purposes of Emeritus status contribution means:
+* Opening or reviewing a PR to any repo in the Strimzi organization.
+* Participating in discussions with the Strimzi community in any of its venues (e.g. Slack, Email, Community call, etc.) This includes non-public participation (e.g. on the maintainers-only email list)
```

Comment:
Some thoughts on this:
* This will be very hard to track because there is no simple way to check the 6 months of no engagement, especially by including the _private_ channels. I do not think you can easily query it. Maybe if it was only GitHub, you could see it from DevStats. But not when you include all mailing lists, Slack etc.
* The emeritus maintainers re-activate themselves by any voting, any PR approval, or any other act like that. So the way I see it, the only thing it really does, is that it bypasses the ⅔ majority votes. Is that the intention? If yes, would it make sense to think about how to change the ⅔ majority vote?

Some ideas I would like to throw in for consideration:
* Not being in touch with the project for over 6 months means that the person would be out of touch with the code,    our users, etc. It does not make sense to me to have the emeritus maintainers re-instantiate themselves by doing something. I think there should be some process to re-activate the emeritus maintainer. For example, having a vote with at least three +1 votes / no -1 votes to re-instantiate the emeritus maintainer as an active maintainer.
* I was also thinking about the criteria ... but it is hard to come up with some objective measure that would make sense for everyone and be easy to measure. I wonder if it would make sense to move people to emeritus simply by a vote with 3 +1 votes as a criteria. That way, you can solve the issues with ⅔ majority votes and have a better (although more subjective) way to decide who is or isn't active. But to be honest, I don't like it either, so I guess it would be fine to keep the current suggestions.

Reply:
> This will be very hard to track because there is no simple way to check the 6 months of no engagement, especially by including the private channels. I do not think you can easily query it. Maybe if it was only GitHub, you could see it from DevStats. But not when you include all mailing lists, Slack etc.

Both the CNCF and Apache Kafka links I provided as example are general around it. I think it's pretty visible on mailing lists and Slack when someone is not active anymore (despite you don't have some DevStats tool like for GitHub). It's pretty clear to me who hasn't been active since 6 months and more right now. I don't think we should make the process complicated. Or at least what are you proposing to do so?

> The emeritus maintainers re-activate themselves by any voting, any PR approval, or any other act like that. So the way I see it, the only thing it really does, is that it bypasses the ⅔ majority votes. Is that the intention? If yes, would it make sense to think about how to change the ⅔ majority vote?

The intention is that they should show the intention to restart contributing. Of course, for me, just one PR, one review, one <anything> is not enough, I would like to see consistency. Using the voting comes back to the original main issue about the 2/3.

> Not being in touch with the project for over 6 months means that the person would be out of touch with the code, our users, etc. It does not make sense to me to have the emeritus maintainers re-instantiate themselves by doing something. I think there should be some process to re-activate the emeritus maintainer. For example, having a vote with at least three +1 votes / no -1 votes to re-instantiate the emeritus maintainer as an active maintainer.
I was also thinking about the criteria ... but it is hard to come up with some objective measure that would make sense for everyone and be easy to measure. I wonder if it would make sense to move people to emeritus simply by a vote with 3 +1 votes as a criteria. That way, you can solve the issues with ⅔ majority votes and have a better (although more subjective) way to decide who is or isn't active. But to be honest, I don't like it either, so I guess it would be fine to keep the current suggestions.

Kind of related to the previous question. Yes, voting could be a way but using the "proposal" system (3 + 1 votes) not based on the 2/3 pool otherwise same issue again. Of course, starting a vote (by any of the other maintainers) should be based on a consistency contribution back by the emeritus maintainer.

Anyway I am confused now, because you are proposing the voting system but then "I don't like it either, so I guess it would be fine to keep the current suggestions.". :-)

Reply:
TBH, I have no idea how can you say someone was not active on Slack for 6 months given how Slack threading etc. works.

I do not feel too strongly about the process for moving people to emeritus status. If you want to somehow keep tracking the six months, I can live with it. I feel a bit more strongly about the getting back to regular maintainer only through vote.

Reply:
wouldn't make sense that emeritus maintainer itself could start the vote somehow? Not sure if emeritus maintainers will still be part of maintainers mailing lists, channels, etc.

Reply:
> Not sure if emeritus maintainers will still be part of maintainers mailing lists, channels, etc.

I guess that is an interesting point. I assume they would remain part of it in Paolo's proposal as any sentence they speak there would make them active again. And I'm not sure I see any reason to kick them out as even emeritus maintainer can provide advice for example without being directly involved with the code or having interest to vote on things etc.

As for my idea with the vote, I do not have a problem if they start it. But they can also easily ask an active maintainer to do it, I'm not sure I see much difference one way or another (you need at least 3 +1 votes - one of the +1 people surely can start the vote for you if needed).

Reply:
An Emeritus Maintainer has, by definition, been a Maintainer in the past. IMO it would make more sense for them to take an active action to become a maintainer again. eg. Raise a PR for a vote which marks them as active again. They know what's expected of that role and may or may not be in a position to undertake it again. 

However, I think that Emeritus Maintainers should be encouraged to contribute as and when they can, without  automatically making them an active maintainer. They should be able to operate as any other member of the community without fear that they are doing something which would commit them to more involvement if they're not able to. If anything, they should be encouraged to be as active as they feel able to be.

Reply:
I agree on using a voting process to have the emeritus to be consider active again (taking into account that he could back to be emeritus if no contributions within 6 months). We should just pay attention if an emeritus would like to come back being active to count for any voting he/she knows is coming (i.e. new maintainer) but then back to not contributing and being removed after 6 months again. This would be a pretty bad behaviour.
I also agree on having an emeritus maintainer access on any Strimzi channel to discuss, raise idea, contribute in any way which doesn't implies long term commitment.
I updated the proposal accordingly.